src/repositories/SlotRepository.ts

```diff
@@ -1,12 +1,37 @@
-import { Repository } from "typeorm";
 import Database from "../providers/Database";
 import Slot from "../entities/Slot";
+import ISlotRepository from "../types/Slot/ISlotRepository";
 
-const slotRepository = (): Repository<Slot> => Database.getConnection()
+const slotRepository = (): ISlotRepository => Database.getConnection()
   .getRepository(Slot).extend({
+    disableSlotsByRoomId(roomId: number): Promise<void> {
+      return this.createQueryBuilder()
+        .update(Slot)
+        .set({ isDisabled: true })
+        .where(
+          "roomId = :roomId", { roomId },
+        )
+        .execute();
+    },
+
+    findAllSlotsByRoomId(roomId: number): Promise<Slot[]> {
+      return this.find({
+        relations: {
```

Comment:
Yes, I need it, I had tried the query without that relation loaded and it simple doesn't return anything without it because the parameter is meant to equal a relation field.